core/services/run_manager.go

```diff
@@ -109,47 +92,25 @@ func NewRun(
 	orm *orm.ORM,
 	now time.Time) (*models.JobRun, []*adapters.PipelineAdapter) {
 
-	run := models.JobRun{
-		ID:             models.NewID(),
-		JobSpecID:      job.ID,
-		CreatedAt:      now,
-		UpdatedAt:      now,
-		Initiator:      *initiator,
-		InitiatorID:    initiator.ID,
-		TaskRuns:       make([]models.TaskRun, len(job.Tasks)),
-		Status:         models.RunStatusInProgress,
-		CreationHeight: utils.NewBig(currentHeight),
-		ObservedHeight: utils.NewBig(currentHeight),
-		RunRequest:     *runRequest,
-		Payment:        runRequest.Payment,
-	}
-
+	run := models.MakeJobRun(job, now, initiator, currentHeight, runRequest)
 	runAdapters := []*adapters.PipelineAdapter{}
-	for i, task := range job.Tasks {
-		adapter, err := adapters.For(task, config, orm)
-		if err != nil {
-			run.SetError(err)
-			break
-		}
 
-		runAdapters = append(runAdapters, adapter)
+	if currentHeight != nil {
```

Comment:
This seems to change the behaviour in a subtle way... what's the reasoning for this?

Reply:
Ooh, you're right, this drops the list of adapters, I'll revert, will add a test too.

Reply:
I am sad that no tests caught this.